### DISCUSSION TITLE
Support for 32-bit Alpine Linux / iSH on iOS?

### DISCUSSION THREAD

**Discussion by arannya** [coder/code-server#2261] [Q&A]
Recently I've taken a liking to iSH Shell for iOS which basically gives you access to a minimalistic 32-bit Alpine Linux shell on iOS. I tried to install code-server in several different ways but the only method that succeeds is via Yarn albeit with dependency issues. Manually resolving the dependencies allows you to start code-server but trying to access triggers a segmentation fault. Has anyone got farther than this? I should mention that Alpine uses musl C libraries so directly making code-server is not possible.

**Comment by ZeeMoros** [comment]
I have a code-server Alpine 3.12 container that runs correctly, installed with npm :

```dockerfile
RUN apk add --no-cache alpine-sdk bash libstdc++ libc6-compat npm libx11-dev libxkbfile-dev libsecret-dev
RUN npm install -g --cache /tmp/npm-cache --unsafe-perm code-server && rm -r /tmp/npm-cache
```

If that doesn't do the trick the problem runs deeper than dependencies I suppose.

**Comment by sr229** [comment]
As of 2020, 32-bit builds of Node.js are no longer supported officially, this also means x86 builds is not guranteed compatibility anymore. I'm sad to say, you must use a cloud service to run code-server instead.

**Comment by seasonyuu** [comment]
try x86 version in the release 3.5.0, that is the last version build with x86

**Comment by woensug-choi** [comment]
No solution found?

**Reply by jsjoeio** [reply]
have you tried the other solutions mentioned?

**Comment by lg-nineoneninethreenine** [comment]
Do you have a Linux or a Mac device?

**Reply by lg-nineoneninethreenine** [reply]
You can run code-server on that or if you use Windows, use Debian WSL. It works mostly on Debian only.

**Reply by lg-nineoneninethreenine** [reply]
Then, try doing some methods to access code-server through your WiFi, like IP address, or some sort.

**Comment by lg-nineoneninethreenine** [comment]
I don't think you need iSH, because, I found another method. Click the link [here](https://youtu.be/UFxVvrrufsA) which will open a YT video. In that, he CHANGES the config. So, if you have Windows, get Debian with Curl in it or if you use Mac or Linux, copy the code-server curl line, put it in the Linux/Mac Terminal, and wait. After it's installed, run code-server. Then, it will show some line for a config file. Open your local VSCode, and locate and then go to the file(if you don't know how to do on WSL, get Remote-WSL extension, and connect to your distro. Then you are ready to go to the file :+1: ). Then, open your terminal and put 

```bash
ifconfig
```

Now, find your IP address as of WiFi, and paste it in the config on the `bind_addr` line, and replace it from 127.0.0.1:8888 to yourip:8888, where yourip is your IP address as of WiFi. Then, put that link on your iPad and you are good to go :+1: :)
